keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -130,8 +130,17 @@ class ClusterHandler(
       "delayBeforeDisableSec" to 0,
       "delayBeforeScaleDownSec" to 0,
       "maxRemainingAsgs" to 2,
+      // the 2hr default timeout sort-of? makes sense in an imperative
+      // pipeline world where maybe within 2 hours the environment around
+      // the instances will fix itself and the stage will succeed. Since
+      // we are telling the red/black strategy to roll back on failure,
+      // this will leave us in a position where we will instead keep
+      // reattempting to clone the server group because the rollback
+      // on failure of instances to come up will leave us in a non
+      // converged state...
+      "stageTimeoutMs" to Duration.ofMinutes(30).toMillis(),
```

Comment:
Can we make this configurable?

Reply:
this is tucked within a whole block of "lets make the parameters to the deploy strategy configurable". I definitely forsee this being something we could override but I didn't go there first pass